core/src/main/java/org/mskcc/cbio/portal/dao/DaoCancerStudy.java

```diff
@@ -622,6 +621,37 @@ public static void purgeUnreferencedRecordsAfterDeletionOfStudy() throws DaoExce
         }
     }
 
+    /**
+     * delete generic assay meta records if meta is not shared with other studies
+     * @throws DaoException
+     */
+    public static void deleteGenericAssayMeta(int internalCancerStudyId) throws DaoException {
+        String[] deleteGenericAssayStatements = {
+                "DELETE FROM generic_entity_properties WHERE GENETIC_ENTITY_ID IN (SELECT GENETIC_ENTITY_ID FROM genetic_alteration WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=?))",
+                "DELETE FROM genetic_entity WHERE ID IN (SELECT GENETIC_ENTITY_ID FROM genetic_alteration WHERE GENETIC_PROFILE_ID IN (SELECT GENETIC_PROFILE_ID FROM genetic_profile WHERE CANCER_STUDY_ID=? AND GENETIC_ALTERATION_TYPE='GENERIC_ASSAY'))"
+                };
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoCancerStudy.class);
+            if (DaoGenericAssay.shouldGenericAssayMetaBeDeleted(internalCancerStudyId)) {
```

Comment:
Shouldn't this test be made inside DaoCancerStudy.deleteCancerStudy() to avoid this call entirely?  (note, I think nested calls into the dao classes are problematic and I would avoid them if possible).

Reply:
Do you think it's better to check at here: `DaoCancerStudy.deleteGenericAssayMeta`?